src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.PrerenderingState.cs

```diff
@@ -15,6 +15,16 @@ internal partial class EndpointHtmlRenderer
 {
     private static readonly object InvokedRenderModesKey = new object();
 
+    public async ValueTask<IHtmlContent> PrerenderServerPersistedStateAsync()
```

Comment:
Why do we need to go through IHtmlContent for this? Seems like unnecessary overhead.

Reply:
I don't think I understand your question, why do you think this might be unnecessary overhead? This is similar to the existing [ValueTask IHtmlContent PrerenderPersistedStateAsync(HttpContext httpContext, PersistedStateSerializationMode serializationMode)](https://github.com/dotnet/aspnetcore/pull/50373/files/7bbfde083b1f483cd2fc3686cbec1e1f41420281#diff-ed60d57e506a39f7e43eacf616f877964403dba2bc6626a7638e2cbee1a41057R28-R29)

Reply:
Couldn't we just write directly to the response?